src/NHibernate.Test/DebugSessionFactory.cs

```diff
@@ -43,14 +43,13 @@ public partial class DebugSessionFactory : ISessionFactoryImplementor
 		/// it debug or not.
 		/// </summary>
 		public DebugConnectionProvider DebugConnectionProvider
-			=> _debugConnectionProvider ??
-				(_debugConnectionProvider = ActualFactory.ConnectionProvider as DebugConnectionProvider);
+			=> _debugConnectionProvider ??= ActualFactory.ConnectionProvider as DebugConnectionProvider;
 		public ISessionFactoryImplementor ActualFactory { get; }
 
 		public EventListeners EventListeners => ((SessionFactoryImpl)ActualFactory).EventListeners;
 
 		[NonSerialized]
-		private readonly ConcurrentBag<ISessionImplementor> _openedSessions = new ConcurrentBag<ISessionImplementor>();
+		private readonly ConcurrentQueue<ISessionImplementor> _openedSessions = new();
```

Comment:
I finally reproduced the trouble in debug mode, but still did not get more information about it. The unclosed session was the 51st in the bag among 160, but that is an unordered collection, so that does not mean much. With a queue, it will be possible to reliably know if that is the last one of the test or not.